### PR TITLE
Add `traverseWithKey_` example to `foldrWithKey` docs

### PR DESCRIPTION
**Pull request by sjakobi**
Context: https://github.com/haskell-unordered-containers/unordered-containers/pull/108

### CODE REVIEW

File: Data/HashMap/Internal.hs [hunk 1]
```diff
@@ -1848,6 +1848,16 @@ foldl f = foldlWithKey (\a _k v -> f a v)
...
+--
+-- === __Example: @traverseWithKey_@__
+--
+-- 'foldrWithKey' can be used to define a variant of 'Data.Foldable.traverse_'
+-- that has access to the keys:
+--
+-- @
+-- traverseWithKey_ :: 'Applicative' f => (k -> v -> f a) -> 'HashMap' k v -> f ()
+-- traverseWithKey_ f = 'foldrWithKey' (\\k v m -> f k v *> m) (pure ())
+-- @
```
Comment (sjakobi) [inline Data/HashMap/Internal.hs:1860]: I should also an example that uses this function

**Comment by treeowl** [comment]
Hrmph. Not sure if it's better to have this or the actual function. This way is ... a bit documentation bloat. So maybe the other way?

**Comment by sjakobi** [comment]
> Hrmph. Not sure if it's better to have this or the actual function. This way is ... a bit documentation bloat. So maybe the other way?

Yeah, I believe we should rather make this a proper export. Even the definition with  `foldrWithKey` is a bit too non-trivial for my taste.

**Comment by sjakobi** [comment]
Closing in favour of #108.

**Pull request closed without merging**